artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/connect/mirror/AMQPMirrorControllerSource.java

```diff
@@ -281,6 +300,15 @@ public void sendMessage(Transaction tx, Message message, RoutingContext context)
             return;
          }
 
+         if (snfQueue.getPagingStore().page(message, tx, pagedRouteContext, this::copyMessageForPaging)) {
+            return;
+         }
```

Comment:
Before the message would just pile up in the mememory, ignore paging and be ot of order.

if not paging ,this is NoOp. If it paged, I must not route.

Reply:
ok...but if it is paging, it seems like this can throw, or delete the message, and other stuff...is that expected?

Reply:
If an user misconfigured to throw.. yes.. wrong things can happen.


I could log a warn.

Reply:
I'm adding an option to enforce a certain page policy. When sync = true, I'm enforcing BLOCK, and when sync = false, I'm enforcing PAGE. (No drop or fail ever allowed on mirroring).

And if sync we must not page otherwise the CompletionContext is lost through paging.